src/dtos/signup-dto.ts

```diff
@@ -0,0 +1,8 @@
+import * as yup from "yup";
+
+export default yup.object().shape({
+  firstname: yup.string().trim().required(),
+  lastname: yup.string().trim().required(),
```

Comment:
because the string fields in the database right now are supporting up to 191 characters, maybe we can add this validation here like `firstname: yup.string().trim().required().max(191, 'First name can't contain more than 191 characters'` (you can round the number to a prettier one if you wih)